apps/mobile-wallet/src/navigation/SendNavigation.tsx

```diff
@@ -39,18 +41,35 @@ const SendStack = createStackNavigator<SendNavigationParamList>()
 
 const SendNavigation = () => (
   <SendContextProvider>
-    <SendStack.Navigator
-      screenOptions={{
-        headerShown: false // Header is in parent above
-      }}
-      initialRouteName="DestinationScreen"
-    >
-      <SendStack.Screen name="DestinationScreen" component={DestinationScreen} />
-      <SendStack.Screen name="OriginScreen" component={OriginScreen} />
-      <SendStack.Screen name="AssetsScreen" component={AssetsScreen} />
-      <SendStack.Screen name="VerifyScreen" component={VerifyScreen} />
-    </SendStack.Navigator>
+    <HeaderContextProvider>
```

Comment:
Nice! I like that the context gets created only at the very place it's being needed!